src/pages/index.js

```diff
@@ -73,35 +102,145 @@ function handleImageClick(title, link) {
 }
 
 function makeCard(cardData) {
-  const card = new Card(cardData, cardTemplate, handleImageClick);
+  const card = new Card(
+    cardData,
+    cardTemplate,
+    handleImageClick,
+    handleDeleteClick,
+    handleCardLike
+  );
   return card.generateCard();
 }
 
+function checkIfLiked(ifLiked, element) {
+  if (ifLiked === true) {
+    element
+      .querySelector(".heart-button")
+      .classList.toggle("heart-button_active");
+  }
+}
+
+function handleDeleteClick(cardId) {
+  console.log(cardId);
+  deleteConfirmModal.open(cardId);
+}
+
 // VALIDATION
 const editFormValidator = new FormValidator(
   validationSettings,
   profileEditForm
 );
 const cardFormValidator = new FormValidator(validationSettings, cardAddForm);
+const avatarFormValidator = new FormValidator(
+  validationSettings,
+  avatarEditForm
+);
 
+avatarFormValidator.enableValidation();
 editFormValidator.enableValidation();
 cardFormValidator.enableValidation();
 
 /* EVENT HANDLERS */
 
 function handleProfileEditSubmit({ name, badge }) {
-  user.setUserInfo(name, badge);
-  editProfileModal.close();
+  editProfileModal.setLoading(true, "Saving...");
+  api
+    .updateProfile({ name, badge })
+    .then((message) => {
+      console.log(message);
+      user.setUserInfoOnSubmit(name, badge);
+      editProfileModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+      editProfileModal.setLoading(false, "Save");
```

Comment:
`      editProfileModal.setLoading(false, "Save");
`

Please, don't duplicate `.setLoading(false, "Save")` in block `catch` if you have it in `finally`